server/client.go

```diff
@@ -3493,8 +3493,15 @@ func isReservedReply(reply []byte) bool {
 	if isServiceReply(reply) {
 		return true
 	}
+	rLen := len(reply)
 	// Faster to check with string([:]) than byte-by-byte
-	if len(reply) > gwReplyPrefixLen && string(reply[:gwReplyPrefixLen]) == gwReplyPrefix {
+	if rLen > jsAckPreLen+1 && string(reply[:jsAckPreLen]) == jsAckPre {
+		// Not sure what should happen if a JS message is relayed by a client (probably shouldn't support that)
+		// But it should never contain an @ (used for renaming)
+		if strings.Contains(string(reply[jsAckPreLen:]), "@") {
```

Comment:
Just use bytes.Contains imo.

Reply:
I'll deny the prefix, that covers this as well.

Reply:
Do we care if this is under `$JS.ACK.`? I mean maybe we just check that there is no `@` allowed regardless of prefix?

Reply:
we only look for an `@` if the reply subject has the `jsAckPre`

```
	var remapped bool
	if len(creply) > 0 &&
		c.kind != CLIENT && c.kind != SYSTEM && c.kind != JETSTREAM && c.kind != ACCOUNT &&
		bytes.HasPrefix(creply, []byte(jsAckPre)) {
		// We need to rewrite the subject and the reply.
		if li := bytes.LastIndex(creply, []byte("@")); li != -1 && li < len(creply)-1 {
			remapped = true
			subj, creply = creply[li+1:], creply[:li]
		}
	}
```

Reply:
Yes, I wrote that comment before you updated that we would reject JSACK regardless of presence of '@'. So I am ok with current code.